src/SymbolPaths/PersistentStorageManager.cpp

```diff
@@ -2,17 +2,24 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "SymbolPaths/QSettingsWrapper.h"
+#include "SymbolPaths/PersistentStorageManager.h"
 
 #include <QSettings>
 #include <filesystem>
+#include <memory>
 
 constexpr const char* kSymbolPathsSettingsKey = "symbol_directories";
 constexpr const char* kDirectoryPathKey = "directory_path";
 
 namespace orbit_symbol_paths {
 
-std::vector<std::filesystem::path> LoadPaths() {
+class PersistentStorageManagerImpl : public PersistentStorageManager {
```

Comment:
Is it really desired to hide the implementation?

Have you considered creating a `QSettingsBasedStorageManager` class that implements the `PersistentStorageManager` interface in a separate header?

Reply:
I'm slightly opposed to the requirement of wrapping this into a unique_ptr for no reason.

Reply:
done in #3347